kernel/tests/dv.rs

```diff
@@ -1,12 +1,32 @@
 //! Read a small table with/without deletion vectors.
 //! Must run at the root of the crate
+use std::ops::Add;
 use std::path::PathBuf;
 
 use delta_kernel::engine::sync::SyncEngine;
-use delta_kernel::Table;
+use delta_kernel::scan::ScanResult;
+use delta_kernel::{DeltaResult, Table};
 
+use itertools::Itertools;
 use test_log::test;
 
+fn count_total_scan_rows(
+    stream: impl Iterator<Item = DeltaResult<ScanResult>>,
+) -> DeltaResult<usize> {
+    stream
+        .map(|sr| {
+            let sr = sr?;
+            let data = sr.raw_data?;
+            // NOTE: The mask only suppresses rows for which it is both present and false.
+            let deleted_rows = sr
```

Comment:
just some rename suggestions
```suggestion
fn count_total_scan_rows(
    scan_result_iter: impl Iterator<Item = DeltaResult<ScanResult>>,
) -> DeltaResult<usize> {
    scan_result_iter
        .map(|scan_result| {
            let scan_result = scan_result?;
            let data = scan_result.raw_data?;
            // NOTE: The mask only suppresses rows for which it is both present and false.
            let deleted_rows = scan_result
```